.platform.app.yaml

```diff
@@ -11,51 +11,21 @@ type: nodejs:12
 
 # Use Yarn instead of npm.
 build:
-    flavor: none
+  flavor: none
 
-# Build dependencies. 
+# Build dependencies.
 dependencies:
-    nodejs:
-        yarn: "1.22.5"
+  nodejs:
+    yarn: "1.22.5"
 
 # The hooks executed at various points in the lifecycle of the application.
 hooks:
-    build: |
-        # Download dependencies and build Strapi.
-        yarn
-        yarn build
-
-        # Local development will commit changes to subdirectories that will become mounts on
-        # Platform.sh and overwritten during deployments. To prevent that, set aside during builds.
-        if [ -n "$(ls -A extensions)" ]; then
-            mkdir extensions-tmp && mv extensions/* extensions-tmp
-        fi        
-        if [ -n "$(ls -A api)" ]; then
-            mkdir api-tmp && mv api/* api-tmp
-        fi
-        if [ -n "$(ls -A exports)" ]; then
-            mkdir exports-tmp && mv exports/* exports-tmp
-        fi
-        if [ -n "$(ls -A components)" ]; then
-            mkdir components-tmp && mv components/* components-tmp
-        fi
-    deploy: |
-        # Copy overrides back to where they need to be.
-        if [ -n "$(ls -A extensions-tmp)" ]; then
-            cp -r extensions-tmp/* extensions
-        fi        
-        if [ -n "$(ls -A api-tmp)" ]; then
-            cp -r api-tmp/* api
-        fi
-        if [ -n "$(ls -A exports-tmp)" ]; then
-            cp -r exports-tmp/* exports
-        fi
-        if [ -n "$(ls -A components-tmp)" ]; then
-            cp -r components-tmp/* components
-        fi
+  build: |
+    # Download dependencies and build Strapi.
+    yarn
```

Comment:
```suggestion
    yarn --ignore-optional --frozen-lockfile 
```

Reply:
You may no better than me, but it seemed like Yarn only with these flags respected the state of `yarn.lock` during a build.

Reply:
I agree with you, at the end of the day it's really about tradeoffs. I'll make the edit

Reply:
Yarn  --ignore-optional --frozen-lockfile  is actually supposed to be the default yarn behavior, I have no idea why it hasn't been considered, see this: https://github.com/yarnpkg/yarn/issues/4147